### PR TITLE
Replace wgpu-native with wgpu-core

### PR DESCRIPTION
**Pull request by zakorgy**
This addresses `1` form https://github.com/servo/servo/issues/24706#issuecomment-557891841.
cc @jdm @kvark @imiklos

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [ ] These changes fix #___ (GitHub issue number if applicable)

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/25035)

### CODE REVIEW

File: components/script/dom/identityhub.rs [hunk 1]
```diff
@@ -2,46 +2,50 @@
...
         }
     }
 }
 
 #[derive(Debug)]
 pub struct Identities {
-    surface: IdentityManager<SurfaceId>,
+    surface: IdentityManager,
     hub: IdentityHub,
+    backends: [Backend; 3],
```
Comment (kvark) [inline components/script/dom/identityhub.rs]: There is a problem here when multiple backends are involved: we can't use the same `IdentityManager` of, say, devices, for multiple backends. Each backend should have its own set of identity managers, except for `surface` things.

**Review by kvark — CHANGES_REQUESTED** [review]

File: components/script/dom/identityhub.rs (hunk 1 above)
Comment (zakorgy) [reply components/script/dom/identityhub.rs]: Thanks for pointing that out @kvark. I've changed `Identities` to have a hub for each backend in the fixup commit.

**Comment by zakorgy** [comment]
@jdm looks like something went wrong on the CI:
`error: failed to download from https://crates.io/api/v1/crates/fnv/1.0.6/download`

File: components/script/dom/identityhub.rs [hunk 2]
```diff
@@ -2,46 +2,79 @@
...
-            adapters: IdentityManager::new(backend),
-            devices: IdentityManager::new(backend),
+            adapters: IdentityManager::default(),
+            devices: IdentityManager::default(),
+            backend,
         }
     }
+
+    fn create_adapter_id(&mut self) -> AdapterId {
+        self.adapters.alloc::<AdapterId>(self.backend)
```
Comment (kvark) [inline components/script/dom/identityhub.rs]: no need for explicit generic here

**Review by kvark — APPROVED** [review]

File: components/script/dom/identityhub.rs (hunk 2 above)
Comment (zakorgy) [reply components/script/dom/identityhub.rs]: Removed it in the rebase

**Comment by zakorgy** [comment]
@jdm I've rebased the PR

**Comment by kvark** [comment]
> Tests with unexpected results:
  ▶ FAIL [expected PASS] /css/CSS2/bidi-text/line-breaking-bidi-002.xht
  │   → /css/CSS2/bidi-text/line-breaking-bidi-002.xht e5c1b7f0f7a260f869a743f4e2e9afb4351ddf30
  └   → /css/CSS2/bidi-text/line-breaking-bidi-002-ref.xht 436a7f403ec6e1a4cd9f165ea5ecae271b960364
  ▶ OK [expected TIMEOUT] /html/semantics/embedded-content/the-iframe-element/iframe_sandbox_popups_escaping-3.html
  ▶ Unexpected subtest result in /html/semantics/embedded-content/the-iframe-element/iframe_sandbox_popups_escaping-3.html:
  │ FAIL [expected TIMEOUT] Check that popups from a sandboxed iframe escape the sandbox if\n       allow-popups-to-escape-sandbox is used
  │   → assert_equals: It came from a sandboxed iframe expected "null" but got "http://web-platform.test:8000"
  │ 
  │ onmessage</onmessage<@http://web-platform.test:8000/html/semantics/embedded-content/the-iframe-element/iframe_sandbox_popups_escaping-3.html:17:7
  │ Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1984:25
  └ Test.prototype.step_func_done/<@http://web-platform.test:8000/resources/testharness.js:2025:32


doesn't look related

**Comment by jdm** [comment]
@bors-servo retry
* #23290

**Pull request merged**